src/pages/en/reference/directives-reference.md

```diff
@@ -0,0 +1,195 @@
+
+💡 *Useful for immediately-visible UI elements that need to be interactive as soon as possible.*
+
+### `client:idle`
+
+Start importing the component JS as soon as main thread is free.
+
+💡 *Useful for items that don't need to be immediately interactive.*
+
+### `client:visible`
+
+Start importing the component JS as soon as the element enters the viewport.
+
+💡 *Useful for content lower down on the page.*
+
+### `client:media={QUERY}`
+
+Start importing the component JS as soon as the browser matches the given media query.
+
+💡 *Useful for sidebar toggles, or other elements that might be used only for certain screen sizes.*
+
+> ⚠️ Remember, this directive only refers to making the component *interactive* at certain media queries. This does not affect the component being *rendered to the screen*, nor its *visibility*!
+
+### `client:only=" "`
+
+Start importing the component JS at page load, similar to `client:load`.
+
+💡 *Useful for components that are entirely dependent on client-side APIs.* 
+
+>⚠️ This component will be **skipped** at build time, and to assist the client, you should specify which renderer to use from the array in your [`astro.config.mjs` configuration](/en/reference/configuration-reference).
+>
+> e.g. `<client:only="react" />` or `<client:only="my-custom-renderer" />`
+
+## Script and Style tags
+
+These directives can be used on HTML `<script>` and `<style>` tags.
+
+### `define:vars={variables}`
+
+Pass variables into a `<script>` or `<style>` tag. Any *serializable* front matter variable is supported, including props passed to your component through `Astro.props`.
+
+>⚠️ `define:vars` cannot be used on a `<script>` tag that makes use of the `hoist` directive.
+>
+> e.g. `<script hoist type="module" define:vars={{ myVariable }}>` will not work!
+
+```astro
+---
+const foregroundColor = "rgb(221 243 228)";
+const backgroundColor = "rgb(24 121 78)";
+const message = "Astro is awsome!";
+---
+<h1>Hello</h1>
+
+<style define:vars={{ textColor: foregroundColor, backgroundColor }}>
+  h1 {
+    background-color: var(--backgroundColor);
+    color: var(--textColor);
+  }
+</style>
+
+<script define:vars={{ message }}>
+  alert(message);
+</script>
+```
+
+### `hoist`
+
+Opt-in the contents of a `<script>` tag to being bundled and processed. ESM imports work, even to npm packages.
+
+Astro detects these JavaScript client-side imports and then builds, optimizes, and adds the JavaScript to the page automatically.
+
+```astro
+// ESM import
+<script hoist type="module">
+  import './some-external-script.js';
+</script>
+```
+
+>⚠️ `hoist` Always needs to be used in conjunction with `type="module"`.
+>
+> e.g. `<script hoist type="module">`
+
+### `global`
+
+Make the contents of a `<style>` tag apply globaly on pages where the component is included by disabling Astro's CSS scoping system.
+
+This is equivalent to wrapping all of the selectors within a `<style>` tag with `:global()`.
+
+```astro
```

Comment:
The examples here show that the two forms are analogous, but neither of them describes what the styling/CSS effect produced is. This is a case where I think the original page itself has more information than the part on the directives page, so probably requires deciding what is going to go where. (And, I'm not convinced having this here is better than having it on the CSS page itself, making readers click through.)

Reply:
I was originally worried about content duplication, which is why I moved it here. I've added a link back to the styling page - is that good enough for now?